src/gluonnlp/data/transforms.py

```diff
@@ -1011,6 +1057,95 @@ def convert_tokens_to_ids(self, tokens):
         return self.vocab.to_indices(tokens)
 
 
+class BERTSPTokenizer(BERTTokenizer):
+    r"""End-to-end SentencePiece tokenization for BERT models.
+    It works best with BERTSentenceTransform()
+
+    Parameters
+    ----------
+    path : str
+        Path to the pre-trained subword tokenization model.
+    vocab : gluonnlp.Vocab or None, default None
+        Vocabulary for the corpus.
+        If vocab == None, then pre-trained subword tokenization model will be used for indexing.
+    num_best : int, default 0
+        A scalar for sampling subwords. If num_best = {0,1}, no sampling is performed.
+        If num_best > 1, then samples from the num_best results.
+        If num_best < 0, then assume that num_best is infinite and
+        samples from the all hypothesis (lattice) using forward-filtering-and-backward-sampling
+        algorithm.
+    alpha : float
+        A scalar for a smoothing parameter. Inverse temperature for probability rescaling.
+    lower : bool, default True
+        whether the text strips accents and convert to lower case.
+        If you use the BERT pre-training model,
+        lower is set to Flase when using the cased model,
+        otherwise it is set to True.
+    max_input_chars_per_word : int, default 200
+
+    Examples
+    --------
+    >>> url = 'http://repo.mxnet.io/gluon/dataset/vocab/test-0690baed.bpe'
+    >>> f = gluon.utils.download(url, overwrite=True)
+    -etc-
+    >>> sp_tokenizer = BERTSPTokenizer(f, lower=True)
+    >>> sentence = 'This is a very awesome, life-changing sentence.'
+    >>> sp_tokenizer(sentence)
+    ['▁this', '▁is', '▁a', '▁very', '▁awesome', '▁,', '▁life', '▁-', '▁changing', '▁sentence', '▁.']
+
+    """
+
+    def __init__(self,
+                 path,
+                 vocab=None,
+                 num_best=0,
+                 alpha=1.0,
+                 lower=True,
+                 max_input_chars_per_word=200):
+        super(BERTSPTokenizer, self).__init__(vocab, lower,
+                                              max_input_chars_per_word)
+        self._path = path
+        self._num_best = num_best
+        self._alpha = alpha
+        self.sentencepiece = None
+
+    def _activate_sp(self):
+        self.sentencepiece = SentencepieceTokenizer(self._path, self._num_best,
+                                                    self._alpha)
+
+    def _tokenize_wordpiece(self, text):
+        """Tokenizes a piece of text into its word pieces.
+
+        This use Google's SentencePiece tokenizer model file
```

Comment:
This uses

Reply:
> 
> 
> Nice work! Would you mind adding an unit test for this?

I will add.

Reply:
@eric-haibin-lin How can I upload test sentencepiece model for unittest like [test_sentencepiece_tokenizer_subword_regularization](https://github.com/dmlc/gluon-nlp/blob/e2b2766a4e970affc42ff8505a69a51eafea4e19/tests/unittest/test_transforms.py#L140) ?

Reply:
@haven-jeon I can help with that

Reply:
@szha  Thanks. Plz let me know.

Reply:
I uploaded the file to apache-mxnet/gluon/dataset/vocab/test-682b5d15.bpe

Reply:
Thanks, I am working.